src/lib/pages/backoffice/Loan/LoanDetails/LoanUpdateDates/LoanUpdateDates.js

```diff
@@ -0,0 +1,239 @@
+import React, { Component } from 'react';
+import { Button, Divider, Form, Icon, Message, Modal } from 'semantic-ui-react';
+import { DatePicker } from '@components/DatePicker';
+import { fromISO, toShortDate } from '@api/date';
+import { DateTime } from 'luxon';
+import { PropTypes } from 'prop-types';
+import { invenioConfig } from '@config';
+
+export default class LoanUpdateDates extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      modalOpen: false,
+      startDate: '',
+      endDate: '',
+      requestStartDate: '',
+      requestExpireDate: '',
+    };
+  }
+
+  today = () => {
+    return toShortDate(DateTime.local());
+  };
+
+  isCancelled = () => {
+    const {
+      loan: { metadata },
+    } = this.props;
+    return metadata.state === 'CANCELLED';
+  };
+
+  isActiveOrCompleted = () => {
+    const {
+      loan: { metadata },
+    } = this.props;
+    return (
+      invenioConfig.CIRCULATION.loanActiveStates.includes(metadata.state) ||
+      metadata.state === 'ITEM_RETURNED'
+    );
+  };
+
+  handleStartDateChange = value => {
+    this.setState(
+      this.isActiveOrCompleted()
+        ? { startDate: value }
+        : { requestStartDate: value }
+    );
+  };
+
+  handleEndDateChange = value => {
+    this.setState(
+      this.isActiveOrCompleted()
+        ? { endDate: value }
+        : { requestExpireDate: value }
+    );
+  };
+
+  handleOpenModal = () => {
+    const {
+      loan: { metadata },
+    } = this.props;
+    const createDate = str => (str ? toShortDate(str) : null);
+    this.setState({
+      modalOpen: true,
+      // Reset form state when modal is opened
+      startDate: createDate(metadata.start_date),
+      endDate: createDate(metadata.end_date),
+      requestStartDate: createDate(metadata.request_start_date),
+      requestExpireDate: createDate(metadata.request_expire_date),
+    });
+  };
+
+  handleCloseModal = () => {
+    const { clearError } = this.props;
+    clearError();
+    this.setState({ modalOpen: false });
+  };
+
+  isSelectionValid = () => {
+    const {
+      startDate,
+      endDate,
+      requestStartDate,
+      requestExpireDate,
+    } = this.state;
+    const active = this.isActiveOrCompleted();
+    const start = active ? startDate : requestStartDate;
+    const end = active ? endDate : requestExpireDate;
+    return start && end && fromISO(start) < fromISO(end);
+  };
+
+  handleCreateLoan = () => {
+    const {
+      loan: {
+        metadata: { pid },
+      },
+      loanUpdateDates,
+    } = this.props;
+    const {
+      startDate,
+      endDate,
+      requestStartDate,
+      requestExpireDate,
+    } = this.state;
+    const data = this.isActiveOrCompleted()
```

Comment:
this could be actually handled on backend. We send start date and end date and the backoffice knows is it is a request or actual loan, therefore less logic on the frontend. I think it could also reduce the number of if statements on the backend. Is there any reason for not doing it in the backend?

Reply:
The frontend needs to know the state in order to display the correct labels. The backend checks it as well because it is required to.

Reply:
good point!